pkg/infra/log/log.go

```diff
@@ -190,6 +191,32 @@ func (cl *ConcreteLogger) log(msg string, logLevel level.Value, args ...interfac
 	return cl.Log(append([]interface{}{level.Key(), logLevel, "msg", msg}, args...)...)
 }
 
+func (cl *ConcreteLogger) logCtx(ctx context.Context, msg string, logLevel level.Value, args ...interface{}) error {
+	allArgs := append([]interface{}{level.Key(), logLevel, "msg", msg}, args...)
```

Comment:
It is nit, maybe we could enforce only allow log with context later. so we can search logs with trace id all the time.

Reply:
Can you elaborate on 

> enforce only allow log with context later

The `traceID` to be included in logs needs to always come from `context.Context` since it's generated on a per-HTTP-request.